### DISCUSSION TITLE
Where are Copy and Paste key bindings defined?

### DISCUSSION THREAD

**Discussion by Serifini** [WayfireWM/wayfire#2093] [General]
I'm new to Wayland and wayfire and have been working my way through a list of things I want to do regarding configuring it to my tastes. One thing I want to do is to change the key bindings that map to the copy and paste actions. By default the bindings seem to be ctrl shift C and ctrl shift V but I'm at a loss as to where these are actually defined so that I can change them.

I've also found the wl-copy and wl-paste command line tools. Can these be used by wayfire to implement copy and paste functionality? For this I've been experimenting with the command plugin but this doesn't seem the way to go. Does anyone have configuration examples for using these commands to provide general copy and paste services in wayfire?

**Comment by soreau** [comment]
Clients can initiate copy/paste so you'll find the bindings for them in client and toolkit settings. Binding wl-copy isn't very useful since it copies from the cli or stdin, but you can bind wl-paste to paste what has been copied. There's also highlight+middle click to copy/paste, supported by most clients.
